test/unit/conditionals.spec.ts

```diff
@@ -251,4 +251,12 @@ export class LuaConditionalsTests {
         // Assert
         Expect(result).toBe(expected);
     }
+
+    @Test("If dead code after return")
+    public ifDeadCodeAfterReturn(): void {
+        const result = util.transpileAndExecute(
+            `if (true) { return 3; const b = 8; }`);
+
+        Expect(result).toBe(3);
+    }
```

Comment:
switch also use transpileblock irrc, but I think this is less important than the namespace test and probably not needed.